aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SigV4AuthDecorator.kt

```diff
@@ -127,30 +140,39 @@ private class AuthServiceRuntimePluginCustomization(private val codegenContext:
                 val serviceHasEventStream = codegenContext.serviceShape.hasEventStreamOperations(codegenContext.model)
                 if (serviceHasEventStream) {
                     // enable the aws-runtime `sign-eventstream` feature
-                    addDependency(AwsCargoDependency.awsRuntime(runtimeConfig).withFeature("event-stream").toType().toSymbol())
+                    addDependency(
+                        AwsCargoDependency.awsRuntime(runtimeConfig).withFeature("event-stream").toType().toSymbol(),
+                    )
                 }
                 section.registerAuthScheme(this) {
                     rustTemplate("#{SharedAuthScheme}::new(#{SigV4AuthScheme}::new())", *codegenScope)
                 }
+
+                if (codegenContext.serviceShape.supportedAuthSchemes().contains("sigv4a")) {
+                    Attribute.featureGate("sigv4a").render(this)
+                    section.registerAuthScheme(this) {
+                        rustTemplate("#{SharedAuthScheme}::new(#{SigV4aAuthScheme}::new())", *codegenScope)
+                    }
```

Comment:
I think we should make a utility for feature gating that accepts a writable and wraps it in a block so that we know that it's feature gated and if another line gets added later it's still in the feature gate

Reply:
done.